File: php/2744/2744.php
<?php

class Solution
{

    /**
     * @param String[] $words
     * @return Integer
     */
    function maximumNumberOfStringPairs($words)
    {
        $count = 0;
        $mapWords = [];
        foreach ($words as $word) {
            if (isset($mapWords[$word])) {
                $count++;
            }
            $mapWords[strrev($word)] = 1;
        }

        return $count;
    }
}

echo (new Solution())->maximumNumberOfStringPairs(["cd","ac","dc","ca","zz"]);

File: go/2744/2744.go
package _2744

func maximumNumberOfStringPairs(words []string) int {
	var count int
	mapWords := make(map[string]bool)

	for i := 0; i < len(words); i++ {
		if _, ok := mapWords[words[i]]; ok {
			count++
			continue
		}
		mapWords[Reverse(words[i])] = true
	}
	return count
}

func Reverse(s string) string {
	runes := []rune(s)
	for i, j := 0, len(runes)-1; i < j; i, j = i+1, j-1 {
		runes[i], runes[j] = runes[j], runes[i]
	}
	return string(runes)
}
